pkg/asset/installconfig/gcp/validation_test.go

```diff
@@ -105,6 +110,61 @@ var (
 			IpCidrRange: validCIDR,
 		},
 	}
+
+	validDefaultMachineImage = func(ic *types.InstallConfig) {
+		ic.Platform.GCP.DefaultMachinePlatform.OSImage.Name = validImage
+		ic.Platform.GCP.DefaultMachinePlatform.OSImage.Project = projectID
+	}
+	validControlPlaneImage = func(ic *types.InstallConfig) {
+		ic.ControlPlane.Platform.GCP.OSImage.Name = validImage
+		ic.ControlPlane.Platform.GCP.OSImage.Project = projectID
+	}
+	validComputeImage = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.GCP.OSImage.Name = validImage
+		ic.Compute[0].Platform.GCP.OSImage.Project = projectID
+	}
+
+	invalidDefaultMachineImage = func(ic *types.InstallConfig) {
+		ic.Platform.GCP.DefaultMachinePlatform.OSImage.Name = invalidImage
+		ic.Platform.GCP.DefaultMachinePlatform.OSImage.Project = projectID
+	}
+	invalidControlPlaneImage = func(ic *types.InstallConfig) {
+		ic.ControlPlane.Platform.GCP.OSImage.Name = invalidImage
+		ic.ControlPlane.Platform.GCP.OSImage.Project = projectID
+	}
+	invalidComputeImage = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.GCP.OSImage.Name = invalidImage
+		ic.Compute[0].Platform.GCP.OSImage.Project = projectID
+	}
+
+	mismatchedDefaultMachineImageArchitecture = func(ic *types.InstallConfig) {
+		ic.Platform.GCP.DefaultMachinePlatform.OSImage.Name = mismatchedArch
+		ic.Platform.GCP.DefaultMachinePlatform.OSImage.Project = projectID
+		ic.ControlPlane.Architecture = types.ArchitectureARM64
+		ic.Compute[0].Architecture = types.ArchitectureARM64
+	}
+	mismatchedControlPlaneImageArchitecture = func(ic *types.InstallConfig) {
+		ic.ControlPlane.Platform.GCP.OSImage.Name = mismatchedArch
+		ic.ControlPlane.Platform.GCP.OSImage.Project = projectID
+		ic.ControlPlane.Architecture = types.ArchitectureARM64
+	}
+	mismatchedComputeImageArchitecture = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.GCP.OSImage.Name = mismatchedArch
+		ic.Compute[0].Platform.GCP.OSImage.Project = projectID
+		ic.Compute[0].Architecture = types.ArchitectureARM64
+	}
+	missingImageArchitecture = func(ic *types.InstallConfig) {
+		ic.ControlPlane.Platform.GCP.OSImage.Name = "missing-arch"
+		ic.ControlPlane.Platform.GCP.OSImage.Project = projectID
+	}
+
+	marketplaceImageAPIResult = &compute.Image{
+		Architecture: "X86_64",
+	}
+
+	unspecifiedMarketplaceImageAPIResult = &compute.Image{
+		Architecture: "ARCHITECTURE_UNSPECIFIED",
+	}
```

Comment:
Let's not pollute the global namespace. These are only used by `TestValidateMarketplaceImages` so they can be defined there.